rbc/omniscidb.py

```diff
@@ -162,53 +167,70 @@ def get_client_config(**config):
-
-    if 'user' in conf:
-        user = conf['user']
-        if 'name' in user:
-            config['user'] = user['name']
-        if 'password' in user:
-            config['password'] = user['password']
-
-    if 'server' in conf:
-        server = conf['server']
-        if 'host' in server:
-            config['host'] = server['host']
-        if 'port' in server:
-            config['port'] = int(server['port'])
-
-    if 'rbc' in conf:
-        rbc = conf['rbc']
-        for k in ['debug', 'use_host_target']:
-            if k in rbc:
-                config[k] = rbc.getboolean(k)
+    conf_file = None
+    for brand, client_conf_env in [('heavyai', 'HEAVYDB_CLIENT_CONF'),
+                                   ('omnisci', 'OMNISCI_CLIENT_CONF')]:
+        conf_file = os.environ.get(client_conf_env, None)
+        if conf_file is not None and not os.path.isfile(conf_file):
+            print('rbc.omnisci.get_client_config:'
+                  f' {client_conf_env}={conf_file!r}'
+                  ' is not a file, ignoring.')
+            conf_file = None
+        if conf_file is None:
+            conf_file_base = os.path.join('.config', brand, 'client.conf')
+            for prefix_env in ['UserProfile', 'AllUsersProfile', 'HOME']:
+                prefix = os.environ.get(prefix_env, None)
+                if prefix is not None:
+                    fn = os.path.join(prefix, conf_file_base)
+                    if os.path.isfile(fn):
+                        conf_file = fn
+                        break
+        if conf_file is not None:
+            break
+
+    if conf_file is not None:
+        conf = configparser.ConfigParser()
+        conf.read(conf_file)
+
+        if 'user' in conf:
+            user = conf['user']
+            if 'name' in user and 'name' not in caller_config:
+                config['user'] = user['name']
+            if 'password' in user and 'password' not in caller_config:
+                config['password'] = user['password']
+
+        if 'server' in conf:
+            server = conf['server']
+            if 'host' in server and 'host' not in caller_config:
+                config['host'] = server['host']
+            if 'port' in server and 'port' not in caller_config:
+                config['port'] = int(server['port'])
+            if 'dbname' in server and 'dbname' not in caller_config:
+                config['dbname'] = server['dbname']
+
+        if 'rbc' in conf:
+            rbc = conf['rbc']
+            for k in ['debug', 'use_host_target']:
+                if k in rbc and k not in caller_config:
+                    config[k] = rbc.getboolean(k)
+
+    if 'dbname' not in config:
+        version = get_heavydb_version(host=config['host'], port=config['port'])
+        if version[:2] >= (6, 0):
+            if version[:3] == (6, 0, 0) and version_date(version) < 20220301:
+                # TODO: remove this if-block when heavydb 6.0 is released.
+                config['dbname'] = 'omnisci'
```

Comment:
Can you open an issue to track this?

Reply:
Sure: https://github.com/xnd-project/rbc/issues/450